flaml/onlineml/autovw.py

```diff
@@ -0,0 +1,128 @@
+                the progressive validation loss
+            config_oracle_random_seed (int): An integer of the random seed used in ConfigOracle
+            model_selection_mode: A string in ['min', 'max'] to specify the objective as
+                minimization or maximization.
+            cb_coef (float): A float coefficient (optional) used in the sample complexity bound.
+        '''
+        self._max_live_model_num = max_live_model_num
+        self._model_select_policy = model_select_policy
+        self._model_selection_mode = model_selection_mode
+        online_trial_args = {"metric": metric,
+                             "min_resource_lease": min_resource_lease,
+                             "cb_coef": cb_coef,
+                             }
+        # setup the arguments for searcher, which contains the ConfigOracle
+        searcher_args = {"init_config": init_config,
+                         "config_oracle_random_seed": config_oracle_random_seed,
+                         'online_trial_args': online_trial_args,
+                         'space': search_space,
+                         }
+        logger.info("search_space %s", search_space)
+        searcher = ChampionFrontierSearcher(**searcher_args)
+        scheduler = ChaChaScheduler(**scheduler_args)
+        logger.info('scheduler_args %s', scheduler_args)
+        logger.info('searcher_args %s', searcher_args)
+        logger.info('automl_runner_args %s', automl_runner_args)
+        self._trial_runner = OnlineTrialRunner(max_live_model_num=self._max_live_model_num,
+                                               searcher=searcher,
+                                               scheduler=scheduler,
+                                               **automl_runner_args)
+        self._best_trial = None
+        # code for bebugging purpose
+        self._prediction_trial_id = None
+        self._iter = 0
+
+    def predict(self, data_sample):
+        """ Predict on the input example (e.g., vw example)
+        """
+        self._best_trial = self._best_trial_selection()
+        self._y_predict = self._best_trial.predict(data_sample)
+        # code for bebugging purpose
+        if self._prediction_trial_id is None or \
+           self._prediction_trial_id != self._best_trial.trial_id:
+            self._prediction_trial_id = self._best_trial.trial_id
+            logger.info('prediction trial id changed to %s at iter %s %s',
+                        self._prediction_trial_id, self._iter, self._best_trial.result.resource_used)
+        return self._y_predict
+
+    def learn(self, data_sample):
+        """Perform one online learning with the given data sample
+
+        Args:
+            data_sample (vw_example/str/list): one data sample on which the model gets updated
+        """
+        self._iter += 1
+        self._trial_runner.step(self._max_live_model_num, data_sample, (self._y_predict, self._best_trial))
```

Comment:
why do we need to pass this variable again? It's already passed when constructing the trial runner